components/giftbox.vue

```diff
@@ -76,27 +76,72 @@
       loading: true,
       features: [],
       selectedFeature: undefined,
-      seenFeatures: new Set(),
     }),
+    computed: {
+      seenFeatures() {
+        return this.$cookies.get('seenFeatures') || {};
+      },
+      cookieOptions() {
+        return {
+          maxAge: 60 * 60 * 24 * 30,
+        };
+      },
+    },
     created() {
       this.fetchFeatures().then((res) => {
         this.features = res;
+        // sync the seenFeatures with the incomming features from the api call
+        this.syncSeenFeatures();
+        // if there are new features
         if (this.features.length > 0) {
           this.selectedFeature = this.features[0];
-          this.seenFeatures.add(this.selectedFeature.uid);
+          this.addToSeenFeatures(this.selectedFeature.uid);
```

Comment:
Good idea to have descriptive methods to mark features as seen or check if they are seen before, instead of working with the `this.features` object directly throughout the whole component. Makes the code easier to follow by hiding the implementation details 👍